### PR TITLE
AEROGEAR-8015 - Add app edit function

### PR DESCRIPTION
**Pull request by ziccardi**
# Motivation
https://issues.jboss.org/browse/AEROGEAR-8015

# What
- Changed the `update` endpoint so that updating the `appName` is supported
- Added a new `update` method to the DataService class
- Added a new `update app` reducer
- Changed MobileCardView to visualize `spec.name` instead of `metadata.name`
- Changed BreadCrump to display `spec.name` instead of `metadata.name`
- Changed DeleteItemButton to display `spec.name` instead of `metadata.name`
- Added EditItemButton

### CODE REVIEW

**Comment by ziccardi** [comment]
@wei-lee : as suggested in the issue description, the current implementation allows only the app name to be changed and the appIdentifier is in `read-only` mode. It could be easily enabled to allow editing that as well. WDYT?

**Comment by wei-lee** [comment]
> @wei-lee : as suggested in the issue description, the current implementation allows only the app name to be changed and the appIdentifier is in `read-only` mode. It could be easily enabled to allow editing that as well. WDYT?

Yeah, we can leave that for now.

File: ui/src/components/overview/MobileClientCardViewItem.js [hunk 1]
```diff
@@ -28,7 +29,8 @@ const MobileClientCardViewItem = (props) => {
       <Card matchHeight /*accented*/ className="mobile-client-card">
         <CardHeading>
           <DropdownKebab id={app.metadata.name} pullRight className="card-dropdown-kebab">
-            <DeleteItemButton itemType="app" itemName={appName} />
+            <EditItemButton itemName={appName} />
```
[the remark this answers is not in the document]
Comment (ziccardi) [reply ui/src/components/overview/MobileClientCardViewItem.js]: done

**Comment by ziccardi** [comment]
@wei-lee : your comments should be addressed now. Mind giving another look?

**Pull request merged**